aws_lambda_powertools/utilities/parameters/base.py

```diff
@@ -159,6 +163,45 @@ def _get_multiple(self, path: str, **sdk_options) -> Dict[str, str]:
         raise NotImplementedError()
 
 
+def get_transform_method(key: str, transform: str) -> Optional[str]:
+    """
+    Determine the transform method
+
+    Examples
+    -------
+        >>> get_transform_method("key", "any_other_value")
+        'any_other_value'
+        >>> get_transform_method("key.json", "auto")
+        'json'
+        >>> get_transform_method("key.binary", "auto")
+        'binary'
+        >>> get_transform_method("key", "auto")
+        None
+
+    Parameters
+    ---------
+    key: str
+        Only used when the tranform is "auto".
+    transform: str
+        Original transform method, only "auto" will try to detect the transform method by the key
+
+    Returns
+    ------
+    Optional[str]:
+        The transform method either when transform is "auto" then None, "json" or "binary" is returned
+        or the original transform method
+    """
+    if transform != "auto":
+        return transform
+
+    if key.endswith(".json"):
+        return "json"
+    elif key.endswith(".binary"):
+        return "binary"
+    else:
+        return None
```

Comment:
Not critical now, but it would be good to parameterise this across the utility. If we add new transformers in the future, it'd be good to have only one place where we need to make a chance to reduce mistakes.

Reply:
@nmoutschen  - maybe we introduce a constant of all of the supported `transform_value` and they can be used as the extension name for the `auto` mode?

Reply:
Not sure if I follow you here, what would it look like?

Reply:
@nmoutschen  - maybe look at the later version of the code.

Reply:
Got you! :) Looks good to me there!